src/Domain/Orientation.hpp

```diff
@@ -0,0 +1,72 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Orientation.
+
+#pragma once
+
+#include <array>
+#include <iosfwd>
+#include <memory>
+
+#include "Domain/Direction.hpp"
+#include "Domain/SegmentId.hpp"
+
+/*!
+ * \ingroup DomainCreators
+ * \brief Allows one to construct a custom Orientation between two Blocks.
+ *
+ * \usage The user provides `2*VolumeDim` Directions, encoding
+ * the correspondence between the directions in each block.
+ * \tparam VolumeDim the dimension of the blocks.
+ */
+template <size_t VolumeDim>
+class Orientation {
+ public:
+  Orientation() = default;
+  Orientation<VolumeDim>(
+      std::array<Direction<VolumeDim>, VolumeDim> mapped_directions);
+  Orientation<VolumeDim>(
+      const std::array<Direction<VolumeDim>, VolumeDim>& directions_in_host,
+      const std::array<Direction<VolumeDim>, VolumeDim>&
+          directions_in_neighbor);
+  ~Orientation() = default;
+  Orientation(const Orientation<VolumeDim>&) = delete;
```

Comment:
Do we think we'll need to copy an Orientation? I'd be in favor of not allowing it until it's actually necessary.